### PR TITLE
Add download proxy parameter

### PR DESCRIPTION
**Pull request by hskwirblies**

### CODE REVIEW

**Comment by electrical** [comment]
jenkins, test it

**Comment by electrical** [comment]
Could you rebase your PR with the master branch? thanks!

**Comment by hskwirblies** [comment]
Done. CLA was signed on friday.

**Comment by electrical** [comment]
Can one of the admins verify this patch?

**Comment by electrical** [comment]
The CLA is an issue on our side, will get that fixed today.

**Comment by electrical** [comment]
jenkins, test it

File: manifests/params.pp [hunk 1]
```diff
@@ -51,6 +51,9 @@
   # package download timeout
   $package_dl_timeout = 600 # 300 seconds is default of puppet
 
+  # download proxy
+  $proxy_url = ''
```
Comment (electrical) [inline manifests/params.pp]: Could you change to undef and move the value to init.pp ?

File: manifests/plugin.pp [hunk 2]
```diff
@@ -95,9 +95,22 @@
...
   }
 
+  # set proxy by override or parse and use proxy_url from
+  # elasticsearch::proxy_url or use no proxy at all
+  
   if ($proxy_host != undef and $proxy_port != undef) {
     $proxy = " -DproxyPort=${proxy_port} -DproxyHost=${proxy_host}"
-  } else {
+  }
+  elsif ($elasticsearch::proxy_url != '') {
```
Comment (electrical) [inline manifests/plugin.pp]: Please change the check to see if proxy_url is not undef

File: manifests/package.pp [hunk 3]
```diff
@@ -103,12 +103,21 @@
         }
         'ftp', 'https', 'http': {
 
+          if $elasticsearch::proxy_url != '' {
```
Comment (electrical) [inline manifests/package.pp]: Check for undef here as well

**Comment by electrical** [comment]
PR testing fails with these errors:


```
manifests/package.pp:108:double_quoted_strings:WARNING:double quoted string containing no variables
manifests/package.pp:109:variables_not_enclosed:WARNING:variable not enclosed in {}
manifests/package.pp:110:variables_not_enclosed:WARNING:variable not enclosed in {}
manifests/package.pp:112:trailing_whitespace:ERROR:trailing whitespace found
manifests/params.pp:55:empty_string_assignment:WARNING:variable assigned to the empty string
manifests/plugin.pp:104:trailing_whitespace:ERROR:trailing whitespace found
```

**Comment by hskwirblies** [comment]
Done as requested :)

**Comment by electrical** [comment]
Jenkins, test it

**Comment by electrical** [comment]
Thank you very much for the changes. I'll try to get it in with the next release.

**Comment by electrical** [comment]
jenkins, test it

**Comment by hskwirblies** [comment]
My pleasure :)

**Pull request merged**